### PR TITLE
Filter adapters with information from the Supervisor

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Filter the adapters shown with information from the Supervisor if the `hassio` integration is loaded

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/core/ha-config-network.ts [hunk 1]
```diff
@@ -73,7 +74,19 @@ class ConfigNetwork extends LitElement {
   private async _load() {
     this._error = undefined;
     try {
-      this._networkConfig = await getNetworkConfig(this.hass);
+      const coreNetowrk = await getNetworkConfig(this.hass);
```
Comment (bdraco) [inline src/panels/config/core/ha-config-network.ts:77]: Looks like a typo? coreNetwork

**Review by bdraco — CHANGES_REQUESTED** [review]

**Comment by bdraco** [comment]
Thanks for doing this

File: src/panels/config/core/ha-config-network.ts (hunk 1 above)
Comment (ludeeus) [reply src/panels/config/core/ha-config-network.ts:77]: Will fix when I get home again :+1:

Comment (bramkragten) [inline src/panels/config/core/ha-config-network.ts:77]:
```suggestion
      const coreNetwork = await getNetworkConfig(this.hass);
```

File: src/panels/config/core/ha-config-network.ts [hunk 2]
```diff
@@ -73,7 +74,19 @@ class ConfigNetwork extends LitElement {
...
+        const interfaces = new Set(
+          supervisorNetwork.interfaces.map((int) => int.interface)
+        );
+        if (interfaces.size) {
+          coreNetowrk.adapters = coreNetowrk.adapters.filter((adapter) =>
+            interfaces.has(adapter.name)
+          );
+        }
+      }
+      this._networkConfig = coreNetowrk;
```
Comment (bramkragten) [inline src/panels/config/core/ha-config-network.ts:89]:
```suggestion
      this._networkConfig = coreNetwork;
```

File: src/panels/config/core/ha-config-network.ts [hunk 3]
```diff
@@ -73,7 +74,19 @@ class ConfigNetwork extends LitElement {
...
     try {
-      this._networkConfig = await getNetworkConfig(this.hass);
+      const coreNetowrk = await getNetworkConfig(this.hass);
+      if (isComponentLoaded(this.hass, "hassio")) {
+        const supervisorNetwork = await fetchNetworkInfo(this.hass);
+        const interfaces = new Set(
+          supervisorNetwork.interfaces.map((int) => int.interface)
+        );
+        if (interfaces.size) {
+          coreNetowrk.adapters = coreNetowrk.adapters.filter((adapter) =>
```
Comment (bramkragten) [inline src/panels/config/core/ha-config-network.ts:84]:
```suggestion
          coreNetowrk.adapters = coreNetwork.adapters.filter((adapter) =>
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**